proposals/0000-modifiers.rst

```diff
@@ -0,0 +1,214 @@
+    -- rejected: the second use of '%m' has an unknown type
+
+  map :: forall (m :: Multiplicity). (a %m -> b) -> [a] %m -> [b]
+    -- accepted: m has a known type
+
+The syntax (and semantics) for modifiers on patterns and record fields is exactly
+as described in the `linear types proposal`_.
+
+.. _`linear types proposal`: https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0111-linear-types.rst#syntax
+  
+Effect and Interactions
+-----------------------
+* It is expected that the matchability of `#242`_ will have a kind ``Matchability``.
+  Then, users will be able
+  to write ``Int %Many %Matchable -> Bool`` or ``Int %Matchable %Many -> Bool``.
+  The details are left to `#242`_ (assuming this proposal is accepted first).
+  The author of `#242`_, Csongor Kiss, was involved in the conceptualization of
+  this proposal.
+
+* Future modifiers will also seamlessly work with existing ones, where order
+  is not expected to matter (though that would be up to other proposals to
+  spell out).
+
+* The ``%1`` will remain a single lexeme and does not participate with this
+  proposal. We may want more exceptions to the general scheme in the future.
+
+* The key action of this proposal is to carve out a new syntax space, anchored
+  by a prefix occurrence of ``%``. Ideally, there would be few exceptions to
+  the general scheme (but ``%1`` is one such exception). It is possible that
+  future extensions to this idea will be disambiguated before the type checker
+  gets a chance to do its work.
+  
+* This proposal means that ``Int %m -> Bool``, acceptable today as a
+  multiplicity-polymorphic function, would be rejected. The user would need
+  to add a kind annotation to tell us that ``m`` is a multiplicity (and not,
+  say, a matchability, which is also expected to support polymorphism). See
+  an Alternative below for a trick to mitigate this problem.
+
+* This proposal paves the way for future proposals introducing new modifiers.
+  Possible candidates:
+
+  * matchability
+  * injectivity on arrows
+  * ``oneShot``
+  * ``inline``
+  * a replacement for the ``{-# OVERLAPPING #-}`` pragmas. These pragmas
+    have, in my opinion, never really fit in: they change the semantics
+    of the declaration. Pragmas are meant to be hints or instructions
+    to the compiler, not something that changes the meaning of a program
+    and its typing rules.
+  * a mechanism for suppressing warnings over one region of a program,
+    instead of at the module level: ``%(suppress "uni-complete-patterns") (case x of ...)``.
+    This could also be done with a pragma.
+
+* Some other features that have had tortuous and torturous syntax
+  discussions may have enjoyed having the modifier option. For example,
+  this might have been used instead of ``type role`` for role annotations:
+  ``data Tagged (%Nominal t) a = Tagged a``. Or it might have been an
+  alternative for ``-XDerivingStrategies``.
```

Comment:
Another possibility that just came to mind, should we have levity modifiers?

Reply:
We could have levity modifiers, but as levity is nicely captured by the kind system, there seems to be only weak motivation to introduce a new mechanism for them.

Reply:
Another feature that in retrospect seems like a modifier: strictness annotations.